### DISCUSSION TITLE
What is the differences between `unit` and `namespace`?

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#2947] [Q&A] [answered]
Unit and Namespace are too kind of files available to choose when adding New File to the project. Isn't Unit the PascalABC.NET equivalent of C# namespace? So what this Namespace is about?

**Comment by AIexandrKotov** [comment, marked as answer]
No, because Pascal languages is not with a multi-pass compiler (mb Oxygene, idk).

Namespaces allow you to split one assembly into several files (as with {$include}), but at the same time Intellisense will work. However, in the main file you need to specify all the uses and {$includenamespace} directives

All class declarations in this case will work as with `type A = class;` declaration

My old project using namespaces:
https://github.com/AIexandrKotov/Sunko
